core/arch/arm/kernel/tee_ta_manager.c

```diff
@@ -1615,26 +1615,34 @@ TEE_Result tee_uta_cache_operation(struct tee_ta_session *sess,
 
 	switch (op) {
 	case TEE_CACHEFLUSH:
-		l1op = DCACHE_AREA_CLEAN_INV;
-		l2op = L2CACHE_AREA_CLEAN_INV;
-		break;
+		/* clean L1, dsb, flush L2, inval L1, dsb. */
```

Comment:
The comment isn't consistent with the code, both with respect to dsb and the final inval l1.

The ARM document recommended an inval l1 at the end of the sequence. Clean-inval l1 as is done here seems safer though, especially as we don't guarantee that there's no accesses to the memory during these operations.